src/mc.rs

```diff
@@ -1230,3 +1217,1063 @@ pub unsafe fn resize_rust<BD: BitDepth>(
+        h as usize,
+        mx as usize,
+        my as usize,
+        dx as usize,
+        dy as usize,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+// TODO(legare): Temporarily pub until init fns are deduplicated.
+pub(crate) unsafe extern "C" fn prep_bilin_scaled_c_erased<BD: BitDepth>(
+    mut tmp: *mut int16_t,
+    mut src: *const DynPixel,
+    mut src_stride: ptrdiff_t,
+    w: libc::c_int,
+    mut h: libc::c_int,
+    mx: libc::c_int,
+    mut my: libc::c_int,
+    dx: libc::c_int,
+    dy: libc::c_int,
+    bitdepth_max: libc::c_int,
+) {
+    prep_bilin_scaled_rust(
+        tmp,
+        src.cast(),
+        src_stride as usize,
+        w as usize,
+        h as usize,
+        mx as usize,
+        my as usize,
+        dx as usize,
+        dy as usize,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+// TODO(legare): Temporarily pub until init fns are deduplicated.
+pub(crate) unsafe extern "C" fn avg_c_erased<BD: BitDepth>(
+    mut dst: *mut DynPixel,
+    dst_stride: ptrdiff_t,
+    mut tmp1: *const int16_t,
+    mut tmp2: *const int16_t,
+    w: libc::c_int,
+    mut h: libc::c_int,
+    bitdepth_max: libc::c_int,
+) {
+    avg_rust(
+        dst.cast(),
+        dst_stride as usize,
+        tmp1,
+        tmp2,
+        w as usize,
+        h as usize,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+// TODO(legare): Temporarily pub until init fns are deduplicated.
+pub(crate) unsafe extern "C" fn w_avg_c_erased<BD: BitDepth>(
+    dst: *mut DynPixel,
+    dst_stride: ptrdiff_t,
+    tmp1: *const int16_t,
+    tmp2: *const int16_t,
+    w: libc::c_int,
+    h: libc::c_int,
+    weight: libc::c_int,
+    bitdepth_max: libc::c_int,
+) {
+    w_avg_rust(
+        dst.cast(),
+        dst_stride as usize,
+        tmp1,
+        tmp2,
+        w as usize,
+        h as usize,
+        weight,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+// TODO(legare): Temporarily pub until init fns are deduplicated.
+pub(crate) unsafe extern "C" fn mask_c_erased<BD: BitDepth>(
+    dst: *mut DynPixel,
+    dst_stride: ptrdiff_t,
+    tmp1: *const int16_t,
+    tmp2: *const int16_t,
+    w: libc::c_int,
+    h: libc::c_int,
+    mask: *const uint8_t,
+    bitdepth_max: libc::c_int,
+) {
+    mask_rust(
+        dst.cast(),
+        dst_stride as usize,
+        tmp1,
+        tmp2,
+        w as usize,
+        h as usize,
+        mask,
+        BD::from_c(bitdepth_max),
+    )
+}
+
+// TODO(legare): Temporarily pub until init fns are deduplicated.
+pub(crate) unsafe extern "C" fn w_mask_444_c_erased<BD: BitDepth>(
+    dst: *mut DynPixel,
+    dst_stride: ptrdiff_t,
+    tmp1: *const int16_t,
+    tmp2: *const int16_t,
+    w: libc::c_int,
+    h: libc::c_int,
+    mut mask: *mut uint8_t,
+    sign: libc::c_int,
+    bitdepth_max: libc::c_int,
+) {
+    w_mask_rust(
+        dst.cast(),
+        dst_stride as usize,
+        tmp1,
+        tmp2,
+        w as usize,
+        h as usize,
+        mask,
+        sign != 0,
```

Comment:
I had `debug_assert!(sign == 0 || sign == 1);`s for these casts previously.  I wasn't totally sure they're always `bool`s.

Reply:
Added those asserts back in.